packages/cozy-sharing/src/helpers/contacts.js

```diff
@@ -39,3 +39,17 @@ export const getOrCreateFromArray = async (client, contacts, createContact) => {
     })
   )
 }
+
+export const getContactsFromGroupId = (contacts = [], groupId) => {
+  return contacts.reduce((contactsfromGroup, contact) => {
+    if (
+      (contact.relationships?.groups?.data || [])
+        .map(group => group._id)
+        .includes(groupId)
+    ) {
+      return [...contactsfromGroup, contact]
+    }
+
+    return contactsfromGroup
+  }, [])
```

Comment:
Here again I assume there are only a few contacts on sharings so the impact should be low, but did you check perfs on this?

As this method is called from ShareRecipientsInput, my understanding is that we have a `.map` inside of a `.reduce` inside of a `.map`, so execution time can grow quickly

Reply:
Unfortunately your assumption is not correct. It corresponds to all the contacts of a user 🙁. I'm going to check the performance of this algo and we have a complexity of  O(n^2 * m^2) so the performance is going to be problematic. I'm going to try to get a bigger set of data locally. Its purpose is just to provide additional information, so I'm going to try to migrate it so that it doesn't block the user's view

Reply:
> Don't you mean all the reachable contacts, i.e. with email or url ?

Also unreachable contacts with a group as I do the filter twice into the ShareRecipientsInput

> For the getContactsFromGroupId, isn't it O(n * m)

You a right, the complexity of `getContactsFromGroupId` is not O(n^2 * m^2). My response was missing the context I wanted to call about the map on group inside `ShareRecipientsInput` which was using this function. 

> And arguably, q is supposed to be rather small compared to n

Totally, I carried out tests with 800 contacts divided into 20 groups corresponding to classes with half of the contact unreachable. I have a Mac M1 and with the simulation of a single core slowed down to * 4 we rarely exceed 50ms, so it's probably acceptable